tests/generate_openl3_regression_data.py

```diff
@@ -0,0 +1,414 @@
+new_audio_emb_linear = np.load(new_audio_emb_linear_path)
+reg_image_emb = np.load(reg_image_emb_path)
+reg_image_emb_linear = np.load(reg_image_emb_linear_path)
+new_image_emb = np.load(new_image_emb_path)
+new_image_emb_linear = np.load(new_image_emb_linear_path)
+
+assert np.allclose(reg_audio_emb['timestamps'], new_audio_emb['timestamps'])
+assert np.allclose(reg_audio_emb['embedding'], new_audio_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb['embedding'], new_image_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_audio_emb_linear['timestamps'], new_audio_emb_linear['timestamps'])
+assert np.allclose(reg_audio_emb_linear['embedding'], new_audio_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb_linear['embedding'], new_image_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+
+# Get audio and image embedding paths from video
+reg_audio_emb_path = os.path.join(test_path, 'data', 'regression', 'bento_audio.npz')
+reg_audio_emb_linear_path = os.path.join(test_path, 'data', 'regression', 'bento_audio_linear.npz')
+new_audio_emb_path = os.path.join(output_dir, 'bento_audio.npz')
+new_audio_emb_linear_path = os.path.join(output_dir, 'bento_audio_linear.npz')
+reg_image_emb_path = os.path.join(test_path, 'data', 'regression', 'bento_image.npz')
+reg_image_emb_linear_path = os.path.join(test_path, 'data', 'regression', 'bento_image_linear.npz')
+new_image_emb_path = os.path.join(output_dir, 'bento_image.npz')
+new_image_emb_linear_path = os.path.join(output_dir, 'bento_image_linear.npz')
+
+# Load audio and image embeddings from video
+reg_audio_emb = np.load(reg_audio_emb_path)
+reg_audio_emb_linear = np.load(reg_audio_emb_linear_path)
+new_audio_emb = np.load(new_audio_emb_path)
+new_audio_emb_linear = np.load(new_audio_emb_linear_path)
+reg_image_emb = np.load(reg_image_emb_path)
+reg_image_emb_linear = np.load(reg_image_emb_linear_path)
+new_image_emb = np.load(new_image_emb_path)
+new_image_emb_linear = np.load(new_image_emb_linear_path)
+
+
+assert np.allclose(reg_audio_emb['timestamps'], new_audio_emb['timestamps'])
+assert np.allclose(reg_image_emb['timestamps'], new_image_emb['timestamps'])
+assert np.allclose(reg_audio_emb['embedding'], new_audio_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb['embedding'], new_image_emb['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_audio_emb_linear['timestamps'], new_audio_emb_linear['timestamps'])
+assert np.allclose(reg_image_emb_linear['timestamps'], new_image_emb_linear['timestamps'])
+assert np.allclose(reg_audio_emb_linear['embedding'], new_audio_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
+assert np.allclose(reg_image_emb_linear['embedding'], new_image_emb_linear['embedding'], rtol=1e-05, atol=1e-06, equal_nan=False)
```

Comment:
See previous comment about separating out generation of regression data and regression tests